src/Ocelot/Configuration/Validator/RouteFluentValidator.cs

```diff
@@ -116,15 +118,10 @@ private static bool IsValidPeriod(FileRateLimitRule rateLimitOptions)
 
             var period = rateLimitOptions.Period;
 
-            var secondsRegEx = new Regex("^[0-9]+s");
-            var minutesRegEx = new Regex("^[0-9]+m");
-            var hoursRegEx = new Regex("^[0-9]+h");
-            var daysRegEx = new Regex("^[0-9]+d");
-
-            return secondsRegEx.Match(period).Success
-                   || minutesRegEx.Match(period).Success
-                   || hoursRegEx.Match(period).Success
-                   || daysRegEx.Match(period).Success;
+            return _secondsRegEx.Match(period).Success
+                   || _minutesRegEx.Match(period).Success
+                   || _hoursRegEx.Match(period).Success
+                   || _daysRegEx.Match(period).Success;

```

Comment:
No catching and processing of the `RegexMatchTimeoutException` exception.

Will it result to unhandled exception by app domain?
Wil it be caught by upper context?